### PR TITLE
UCT/IB: Use NUMA distance to calculate remote device latency

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6018/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6020/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@alex-mikheev pls take a look

File: src/ucs/sys/sys.c [hunk 1]
```diff
@@ -334,6 +332,47 @@ ssize_t ucs_read_file(char *buffer, size_t max, int silent,
...
+    va_end(ap);
+
+    if (read_bytes < 0) {
+        /* read error */
+        return UCS_ERR_IO_ERROR;
+    }
+
+    n = strtol(buffer, &tail, 0);
+    if (*tail != '\0' && !isspace(*tail)) {
+        /* parse error */
```
Comment (alex-mikheev) [inline src/ucs/sys/sys.c]: () around condition

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6023/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6030/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6037/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6041/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6059/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@alex-mikheev pls take a look - added an optimizaiton

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6072/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6078/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6084/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6090/ for details (Mellanox internal link).

**Comment by bureddy** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6094/ for details (Mellanox internal link).

**Pull request merged**